File: tests/csharp/csharp_operators/logical_operators.cs
// vybe-test: csharp/csharp_operators/logical_operators
// origin: languages/csharp/tests/csharp/test_csharp_operators.rs

string __buf = "";

void __P(string s) {
    __buf = __buf + s + "\n";
}

void __Pr(string s) {
    __buf = __buf + s;
}

// The final WriteLine contributes a trailing newline that the expected line
// vector never carried, so BOTH forms are accepted.
void __Check(string want) {
    if (__buf != want && __buf != want + "\n") {
        Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
        throw new Exception("assertion failed");
    }
}

__P((true && true).ToString());
__P((true && false).ToString());
__P((false || true).ToString());
__P((false || false).ToString());
__P((!true).ToString());
__Check("True\nFalse\nTrue\nFalse\nFalse");

File: tests/fortran/control_flow/logical_operators.f90
! vybe-test: fortran/control_flow/logical_operators
! origin: languages/fortran/tests/fortran/test_control_flow.rs

program test
    logical :: a, b
    a = .true.
    b = .false.
    if (a .and. .not. b) then
        if (trim("yes") /= "yes") then
    print *, "FAIL: want [yes] got [", "yes", "]"
    stop 1
end if
    end if
end program test
